core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -1839,13 +1845,18 @@ private RecordCursorFactory generateTableQuery(
                     }
 
                     if (orderByKeyColumn) {
+                        metadata.setTimestampIndex(-1);
                         if (model.getOrderByDirectionAdvice().getQuick(0) == QueryModel.ORDER_DIRECTION_ASCENDING) {
                             symbolValueList.sort(Chars.CHAR_SEQUENCE_COMPARATOR);
                         } else {
                             symbolValueList.sort(Chars.CHAR_SEQUENCE_COMPARATOR_DESC);
                         }
                     }
 
+                    if (clearTimestampIndex) {
```

Comment:
Perhaps this one in place too, i feel it would be less confusing. This is after looking where the boolean is being set